### DISCUSSION TITLE
Finding K-Hop Neighbors

### DISCUSSION THREAD

**Discussion by YoavLotem** [pyg-team/pytorch_geometric#5913] [Q&A]
Hey :)

I'm wondering if its possible to utilize the (excellent) PyG library in order to find k-hop neighbors in a large scale graph database.

How does the PyG "neighbor finding" engine works and where can I find the code?

So lets say for example I have a large sparse/dense adjacency matrix and I want to multiply it by itself in order to find 2-hop neighbors of each vertex.  Does PyG has an optimized method in doing so or do it have a smarter way of finding neighbors?

Thanks!

**Comment by LukeLIN-web** [comment]
(https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/utils/subgraph.html)
https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html#torch_geometric.utils.k_hop_subgraph

**Reply by rusty1s** [reply]
Alternatively, you can use `NeighborLoader` with `num_neighbors=[-1] * k`

**Reply by ChanglongZheng** [reply]
I have this question too, trying **k_hop_subgraph**  and finding it helpful.
